test/framework/src/main/java/org/elasticsearch/transport/nio/ESSelector.java

```diff
@@ -46,12 +46,12 @@
 
     final Selector selector;
     final ConcurrentLinkedQueue<NioChannel> channelsToClose = new ConcurrentLinkedQueue<>();
-    final Set<NioChannel> registeredChannels = Collections.newSetFromMap(new ConcurrentHashMap<NioChannel, Boolean>());
 
     private final EventHandler eventHandler;
     private final ReentrantLock runLock = new ReentrantLock();
     private final AtomicBoolean isClosed = new AtomicBoolean(false);
     private final PlainActionFuture<Boolean> isRunningFuture = PlainActionFuture.newFuture();
+    private final Set<NioChannel> registeredChannels = Collections.newSetFromMap(new ConcurrentHashMap<NioChannel, Boolean>());
```

Comment:
`handleClose` has access to a logger in case an exception occurs. So I am hesitant to remove that, as I think it's weird to pass a logger to Channel. And the event handler is kind of where I log about issues.

In terms of implementing `Closeable` I am open to that. But I guess my hesitation is that the `close()` method should only be called from the selector thread. Right now we have `closeAsync` and `closeFromSelector`. Making it `closeAsync` and `close` kind of removes the context that `close` should only be called from the selector.

Let me know what you think.